### PR TITLE
py-xopen: update release 0.8.2 checksum

### PR DESCRIPTION
**Pull request by odoublewen**
Tested on Centos6, gcc 8.2

This update is needed for #12741

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-xopen/package.py [hunk 1]
```diff
@@ -15,6 +15,7 @@ class PyXopen(PythonPackage):
     homepage = "https://github.com/marcelm/xopen"
     url      = "https://pypi.io/packages/source/x/xopen/xopen-0.1.1.tar.gz"
 
+    version('0.8.2', sha256='003749e09af74103a29e9c64c468c03e084aa6dfe6feff4fe22366679a6534f7')
     version('0.5.0', sha256='b097cd25e8afec42b6e1780c1f6315016171b5b6936100cdf307d121e2cbab9f')
     version('0.1.1', '4e0e955546ee6bee4ea736b54623a671')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-xopen/package.py]: New dependencies and Python versions required:
```python
    install_requires=[                                                          
        'bz2file; python_version=="2.7"',                                       
    ],          
    python_requires='>=2.7, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*, <4',            
```

Comment (odoublewen) [reply var/spack/repos/builtin/packages/py-xopen/package.py]: good catch, although these should have been there for 0.5.0 as well.

https://github.com/marcelm/xopen/blob/v0.5.0/setup.py
https://github.com/marcelm/xopen/blob/v0.8.2/setup.py
https://github.com/marcelm/xopen/blob/v0.1.1/setup.py

File: var/spack/repos/builtin/packages/py-xopen/package.py [hunk 2]
```diff
@@ -15,8 +15,11 @@ class PyXopen(PythonPackage):
...
 
     depends_on('py-setuptools', type='build')
-    depends_on('python@2.6:', type=('build', 'run'))
+    depends_on('py-bz2file', type=('build','run'), when='@0.5:')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-xopen/package.py]:
```suggestion
    depends_on('py-bz2file', type=('build','run'), when='@0.5: ^python@:2.8')
```

Comment (odoublewen) [reply var/spack/repos/builtin/packages/py-xopen/package.py]: Ah, I was trying to figure out what was meant by 
```
   install_requires=[                                                          
        'bz2file; python_version=="2.7"',                                       
    ],       
```
I understand now...

**Pull request merged**